### PR TITLE
Dockerize Prefect Server, Agent and Proxy 

### PR DESCRIPTION
**Pull request by d-jeph**
### Issue
In the current setup it's not possible to run prefect, prefect agent and the proxy as containers. This PR adds the necessary docker files to enable running prefect as containers.

### Changes
 Added the following files and folders:
- A `docker_compose` folder with a `docker_compose` file and `env.template` file. The `docker_compose` file has three services: `prefect_proxy`,`server`, and `agent`
- A `Dockerfile` for the proxy python application
- The `env.template` file contains the necessary environment variables for the `docker_compose` file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,24 +38,56 @@ In the cloned repository, run the following commands:
...
+- populate all the variables in the `.env` file
+- Next we need to build the prefect proxy docker image(if were are using local docker image)
+
+To Build and use proxy docker image, run the below command from the root directory:
+
+```
+docker build -f Docker/Dockerfile --build-arg BUILD_DATE=$(date -u +'%Y-%m-%dT%H:%M:%SZ') --build-arg BUILD_VERSION=0.0.1  -t prefect_proxy:latest .
+```
+
+If we don't build the image locally, docker will try to pull the image from dockerhub when we run docker compose.
```
Comment (edodge) [inline README.md:78]: Can we add more context here? Is it bad if it tries to get it from dockerhub?

**Review by edodge — APPROVED** [review]

File: Docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    && /root/.cargo/bin/uv pip install --system --no-cache-dir -r requirements.txt \
+    # Clean up
+    && apt-get autoremove -y \
+    && apt-get clean -y \
+    && rm -rf /var/lib/apt/lists/*
+
+# Copy the proxy directory contents into the container at /app/proxy
+COPY ../proxy /app/proxy
+
+RUN ls -la /app
```
Comment (yvonnegitau) [inline Docker/Dockerfile:38]: We can remove this if not needed

File: Docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+WORKDIR /app
+
+
+COPY --from=build --chown=app_user:app_user /usr/local/lib/python3.10/site-packages /usr/local/lib/python3.10/site-packages
+# This is added to get packages installed in the build stage
+COPY --from=build --chown=app_user /usr/local/ /usr/local/
+COPY --from=build --chown=app_user:app_user ${WORKDIR}/app ${WORKDIR}
+
+
+RUN ls -la && pwd
```
Comment (yvonnegitau) [inline Docker/Dockerfile:66]: We can remove this if not needed

File: Docker/.env.template [hunk 4]
```diff
@@ -0,0 +1,6 @@
+PREFECT_API_URL="http://127.0.0.1:4200/api"
+LOGDIR="logs"
+PREFECT_WORK_QUEUE_NAME="ddp"
+PREFECT_POOL_NAME="ddp"
+DB_CONNECTION_URL=""
+PREFECT_PROXY_API_PORT="4300"
```
Comment (yvonnegitau) [inline Docker/.env.template:6]: Could we add the DB_CONNECTION_URL example here

**Review by yvonnegitau — APPROVED** [review]
Approved with some comments

**Pull request merged**